backend/ventserver/integration/_trio.py

```diff
@@ -312,3 +334,49 @@ async def process_all(
             nursery.start_soon(
                 process_clock, protocol, channel, push_endpoint.clone()
             )
+
+
+async def initialize_states(
+        states: List[Type[betterproto.Message]],
+        protocol: server.Protocol,
+        filehandler: fileio.Handler,
+        all_states: Dict[
+            Type[betterproto.Message], Optional[betterproto.Message]
+        ]
+) -> None:
+    """Initialize state values from state store or default values."""
+    default_init = list()
+    for state in states:
+        try: # Handle fileio errors
+            filehandler.set_props(state.__name__, "rb")
+            await filehandler.open()
+            message = await filehandler.read()
+            protocol.receive.file.input(
+                file.StateData(state_type=state.__name__, data=message)
+                )
+        except OSError as err:
+            print(err)
+        finally:
+            await filehandler.close()
+
+    while True:
+        try: # Handles data integrity and protocol error
+            event = protocol.receive.file.output()
+            if not event:
+                break
+
+            all_states[type(event)] = event
+        except exceptions.ProtocolDataError as err:
+            print(err)
+
+    for state in states:
+        if not all_states[state]:
+            default_init.append(state)
+
+    for state in default_init:
+        if state is mcu_pb.ParametersRequest:
+            all_states[mcu_pb.ParametersRequest] = mcu_pb.ParametersRequest(
```

Comment:
Would it be possible for us to pass in the default initializations as a parameter to `initialize_states`? Perhaps we could just initialize all those states at the start of the function, and then overwrite them with states from any successfully-loaded files? And then states left out of the list/map/etc. of default initializations remain as `None`, i.e. the backend will wait for a value to be provided by the firmware or the frontend.